Orange/data/util.py

```diff
@@ -28,3 +28,27 @@ def scale(values, min=0, max=1):
     if ptp == 0:
         return np.clip(values, min, max)
     return (-minval + values) / ptp * (max - min) + min
+
+
+class ComputeValueWCommon:
+    """This class stuctures compute_value functions that
+    repeat parts of computation for different variables into
+    common and specific parts.
+
+    Attributes:
+        commonfn    Function that performs computation that is common
+                    for multiple variables. All variables that share
+                    computation need to set the same function instance.
+    """
+    commonfn = lambda data: None
+
+    def __call__(self, data, common=None):
+        """Fallback if common parts are not passed."""
+        if common is None:
+            common = self.commonfn(data)
+        return self.specific(common, data)
+
+    def specific(self, common, data):
+        """Given the common computation results, perform variable-specific
+        part of computation and return new variable values."""
+        raise NotImplementedError("specific computation must be implemented")
```

Comment:
Just if you'd care to know, I, personally, find `NotImplementedError` messages like this one 100% useless.